cvmfs/catalog_mgr_ro.cc

```diff
@@ -39,13 +96,43 @@ LoadReturn SimpleCatalogManager::LoadCatalogByHash(
   assert(shash::kSuffixCatalog == effective_hash.suffix);
   const string url = stratum0_ + "/data/" + effective_hash.MakePath();
 
-  std::string tmp;
-
-  FILE *fcatalog = CreateTempFile(dir_temp_ + "/catalog", 0666, "w", &tmp);
-  ctlg_context->SetSqlitePath(tmp);
-  if (!fcatalog) {
-    PANIC(kLogStderr, "failed to create temp file when loading %s",
-          url.c_str());
+  FILE *fcatalog;
+
+  if (useLocalCache()) {
+    std::string tmp_path = local_cache_dir_ + "/"
+                           + effective_hash.MakePathWithoutSuffix();
```

Comment:
Why calling a path in the cache directory `tmp_path`? It makes it look like the path is in the tmp directory.
```suggestion
    std::string cache_path = local_cache_dir_ + "/"
                           + effective_hash.MakePathWithoutSuffix();
```

Reply:
i had been thinking about this but in the end i decided to go with the following:
`tmp_path` is the path to the new catalog (for writeable catalog it is a copy to a temproary file that is editable)
`cache_path` is the path to the catalog in the cache directory. they are not allowed to edit that file.

we can rename it if you have better ideas. but throughout the method this should be consistent

Reply:
update: after the rewrite for comment above, this should be more clear now